src/httpd.erl

```diff
@@ -122,59 +123,78 @@ handle_receive(Socket, Packet, State) ->
+                                        handler_config => HandlerConfig,
+                                        socket => Socket
+                                    },
+                                    handle_request_state(Socket, NewHttpRequest, CleanState);
                                 Error ->
-                                    ?TRACE("Web socket error: ~p", [Error]),
-                                    {close, create_error(?INTERNAL_SERVER_ERROR, {web_socket_error, Error})}
+                                    {close, create_error(?INTERNAL_SERVER_ERROR, Error)}
                             end;
-                        Error ->
-                            Error
-                    end
+                        ws ->
+                            ?TRACE("Protocol is ws", []),
+                            Config = CleanState#state.config,
+                            Path = maps:get(path, HttpRequest),
+                            case get_handler(Path, Config) of
+                                {ok, PathSuffix, EntryConfig} ->
+                                    WsHandler = maps:get(handler, EntryConfig),
+                                    ?TRACE("Got handler ~p", [WsHandler]),
+                                    HandlerConfig = maps:get(handler_config, EntryConfig, #{}),
+                                    case WsHandler:start(Socket, PathSuffix, HandlerConfig) of
+                                        {ok, WebSocket} ->
+                                            ?TRACE("Started web socket handler: ~p", [WebSocket]),
+                                            NewWebSocketMap = maps:put(Socket, WebSocket, CleanState#state.ws_socket_map),
+                                            NewState = CleanState#state{ws_socket_map = NewWebSocketMap},
+                                            ReplyToken = get_reply_token(maps:get(headers, HttpRequest)),
+                                            ReplyHeaders = #{"Upgrade" => "websocket", "Connection" => "Upgrade", "Sec-WebSocket-Accept" => ReplyToken},
+                                            Reply = create_reply(?SWITCHING_PROTOCOLS, ReplyHeaders, <<"">>),
+                                            ?TRACE("Sending web socket upgrade reply: ~p", [Reply]),
+                                            {reply, Reply, NewState};
+                                        Error ->
+                                            ?TRACE("Web socket error: ~p", [Error]),
+                                            {close, create_error(?INTERNAL_SERVER_ERROR, {web_socket_error, Error})}
+                                    end;
+                                Error ->
+                                    Error
+                            end
+                    end;
+                {error, Reason} ->
+                    CleanBufferMap = maps:remove(Socket, BufferMap),
+                    _CleanState = State#state{pending_buffer_map = CleanBufferMap},
```

Comment:
The above two line end up doing nothing. They should be removed if they aren't needed.

Reply:
done.